src/documentation/programming-guide.md

```diff
@@ -1283,7 +1283,225 @@ public class MyCustomDataType {
+```java
+    PCollection<String> items = ...;
+    PCollection<String> sliding_windowed_items = items.apply(
+        Window.<String>into(SlidingWindows.of(Duration.standardMinutes(30)).every(Duration.standardSeconds(5))));
+```
+```py
+{% github_sample /apache/beam/blob/master/sdks/python/apache_beam/examples/snippets/snippets_test.py tag:setting_sliding_windows
+%}
+```
+
+#### Setting session windows
+
+The following example code shows how to apply `Window` to divide a `PCollection` into session windows, where each session must be separated by a time gap of at least 10 minutes:
+```java
+    PCollection<String> items = ...;
+    PCollection<String> session_windowed_items = items.apply(
+        Window.<String>into(Sessions.withGapDuration(Duration.standardMinutes(10))));
+```
+```py
+{% github_sample /apache/beam/blob/master/sdks/python/apache_beam/examples/snippets/snippets_test.py tag:setting_session_windows
+%}
+```
+
+Note that the sessions are per-key — each key in the collection will have its own session groupings depending on the data distribution.
+
+#### Setting a single global window
+
+If your `PCollection` is bounded (the size is fixed), you can assign all the elements to a single global window. The following example code shows how to set a single global window for a `PCollection`:
+
+```java
+    PCollection<String> items = ...;
+    PCollection<String> batch_items = items.apply(
+        Window.<String>into(new GlobalWindows()));
+```
+```py
+{% github_sample /apache/beam/blob/master/sdks/python/apache_beam/examples/snippets/snippets_test.py tag:setting_global_window
+%}
+```
+
+### Time skew, data lag, and late data
+
+In any data processing system, there is a certain amount of lag between the time a data event occurs (the "event time", determined by the timestamp on the data element itself) and the time the actual data element gets processed at any stage in your pipeline (the "processing time", determined by the clock on the system processing the element). In addition, there are no guarantees that data events will appear in your pipeline in the same order that they were generated in various places on the web.
+
+For example, let's say we have a `PCollection` that's using fixed-time windowing, with windows that are five minutes long. For each window, Beam must collect all the data with an _event time_ timestamp in the given window range (between 0:00 and 4:59 in the first window, for instance). Data with timestamps outside that range (data from 5:00 or later) belong to a different window.
+
+However, data isn't always guaranteed to arrive in a pipeline in correct time order, or to always arrive at predictable intervals. Beam tracks a _watermark_, which is the system's notion of when all data in a certain window can be expected to have arrived in the pipeline. Data that arrives with a timestamp after the watermark is considered **late data**.
```

Comment:
I'd scratch `correct` from `time order`. Timestamp order just is what it is, and it isn't what you get in Beam :-)